File: python/mergeLists.py
import json
import sys

gphotos = []
photoids = {}


for ifpref in sys.argv[1:]:
    ifname = ifpref + ('.json' if not '.json' in ifpref else '')
    photos = json.loads(open(ifname).read())
    for p in photos:
        if not p['id'] in photoids:
            gphotos.append(p)
            photoids[p['id']] = 1

print json.dumps(gphotos, indent=4)

File: ruby/mergeLists.rb
require 'json'

gphotos = []
photoids = {}

ARGV.each { |ifname|
  ifname += ".json" if not ifname =~ /\.\w+$/
	photos = JSON.parse(File.read(ifname))
  photos.each { |photo|
    gphotos.push(photo) if not photoids.has_key?(photo['id'])
    photoids[photo['id']] = 1
  }
}
puts JSON.pretty_generate(gphotos)

